src/site/topic2-pointers.rst

```diff
@@ -0,0 +1,247 @@
+    * Also note that your memory address will differ if you run this, and will differ every time you run the program
+
+* The variable name, ``anInt``, is the label for this memory address in this context
+
+
+Pointer Variable
+================
+
+* Instead of simply printing out the memory address, we can store it in a variable for later use
+* We call these variables *pointers* because we can think of them as pointing to a memory address
+* The trick here is to be mindful of the type of data we are storing
+    * If we want to store an integer pointer, we will have an integer pointer type
+
+
+.. code-block:: cpp
+    :linenos:
+    :emphasize-lines: 4
+
+    int myVar = 25;
+
+    // Create a pointer variable
+	int* foo = &myVar;
+	int bar = myVar;
+
+* In the above example, notice the asterisk (``*``) next to the ``int``
+* This means that the data being stored in the variable labelled ``foo`` must be of type integer pointer
+* ``bar`` on the other hand just has type ``int``
+
+
+.. image:: img/cpp_pointer0.png
+   :width: 200 px
+   :align: center
+   :target: https://www.cplusplus.com/doc/tutorial/pointers/
+
+* In the above example, assuming ``myVar`` is stored at address ``1776``
+    * The address is stored in the integer pointer ``foo``
+    * The integer from ``myVar`` is copied to the integer ``bar``
+
+
+Dereferencing
+=============
+
+* If you have a pointer variable, but you do not care about the memory address; you want to know what is stored *at* the memory address, we *dereference*
+
+.. image:: img/cpp_pointer0.png
+   :width: 200 px
+   :align: center
+   :target: https://www.cplusplus.com/doc/tutorial/pointers/
+
+* For example, if you do not care that the integer pointer ``foo`` stores ``1776``, but you care what is at ``1776``
+
+.. code-block:: cpp
+    :linenos:
+    :emphasize-lines: 2,4
+
+    int myVar = 25;
+	int* foo = &myVar;
+	int bar = myVar;
+	int baz = *foo;
+
+* The use of ``*foo`` on line 4 means *dereference*
+* It means, take the memory address stored in the integer pointer variable ``foo``, but tell me what is in that address
+* If we follow the code on line 4, it will execute like this:
+    * ``foo`` is a label for an integer pointer variable containing ``1776``
+    * But the ``*foo`` means, get what is **at** memory address ``1776``
+    * Take the value there (``25``) and copy it to the integer variable ``baz``
+
+
+.. image:: img/cpp_pointer1.png
+   :width: 200 px
+   :align: center
+   :target: https://www.cplusplus.com/doc/tutorial/pointers/
+
+.. warning::
+
+    Note the two different uses of ``*``. (a) ``int* foo`` and (b) ``*foo``. In (a), we are specifying the type to be
+    integer pointer --- it's having an impact on the type, so the asterisk (``*``) is by the type in the variable
+    declaration. With (b) however, we already have an existing pointer variable that we want to dereference, thus the
+    asterisk (``*``) is by the already existing variable.
```

Comment:
That's just a convention. You can still write `int *foo`. IIRC it depends what side of a declaration it is on only.

Reply:
You totally can, and if anything i think what you've said is more common, but i found over the years that putting it next to the type helps with getting the difference between pointers and dereferencing. 

Again, still all stupid to get into this either way.